lusidtools/apps/flush_transactions.py

```diff
@@ -0,0 +1,211 @@
+            curr_batch = []
+            curr_count = 0
+
+        curr_batch.append(txn_id)
+        curr_count = curr_count + len(f"transactionIds={txn_id}&")
+
+    # Create remainder final batch if any remainder exists
+    if curr_batch:
+        batches.append(curr_batch)
+
+    logging.info(
+        f"Batched the transactions into {len(batches)} batches based on the character limit of {maxCharacterCount}"
+    )
+
+    return batches
+
+
+class TxnGetter:
+    def _get_transactions(self, args, page=None):
+        # make API call to LUSID
+        if page is None:
+            return self.transaction_portfolios_api.get_transactions(
+                args.scope,
+                args.portfolio,
+                from_transaction_date=args.start_date,
+                to_transaction_date=args.end_date,
+                limit=5000,
+            )
+        else:
+            return self.transaction_portfolios_api.get_transactions(
+                args.scope,
+                args.portfolio,
+                from_transaction_date=args.start_date,
+                to_transaction_date=args.end_date,
+                limit=5000,
+                page=page,
+            )
+
+    def __init__(self, args):
+        self.args = args
+        api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=args.secrets
+        )
+        self.transaction_portfolios_api = api_factory.build(
+            lusid.api.TransactionPortfoliosApi
+        )
+        self.stop = False
+
+    def __iter__(self):
+        # get first page and get next page token
+        self.txns = self._get_transactions(self.args)
+        self.next_page = self.txns.next_page
+        return self
+
+    def __next__(self):
+        # if there is a next page token get the next page
+        if self.stop:
+            raise StopIteration
+
+        result = self.txns
+        if self.next_page:
+            self.txns = self._get_transactions(self.args, self.next_page)
+            self.next_page = self.txns.next_page
+            return result
+        else:
+            self.stop = True
+            return result
+
+
+def get_paginated_txns(args):
+    """
+    Gets paginated transactions in a given time window
+
+    Parameters
+    ----------
+    args : Namespace
+        The arguments taken in when command is run
+
+    Returns
+    -------
+    List of all responses from LUSID, each response relating to a page of transactions
+
+    """
+    txn_getter = TxnGetter(args)
+    return [response for response in txn_getter]
+
+
+def get_all_txns(args):
+    """
+    Gets all the transactions in a given time window
+
+    Parameters
+    ----------
+    args : Namespace
+        The arguments taken in when command is run
+
+    Returns
+    -------
+    List of all transactions
+
+    """
+    flat_list = []
+    # Iterate through the outer list
+    for element in get_paginated_txns(args):
+        flat_list = flat_list + element.values
+    return flat_list
```

Comment:
You can try this one on for size if you like `flat_list = [item for sublist in t for item in sublist]` 

From here: https://stackoverflow.com/questions/952914/how-to-make-a-flat-list-out-of-a-list-of-lists